### PR TITLE
Feature/project portfolio ranking

### PR DESCRIPTION
**Pull request by destacey**
## Summary

- Adds portfolio-level project ranking so every project has a required rank within its portfolio.
- Adds a portfolio **Ranking** tab for rank-ordered Proposed, Approved, and Active projects.
- Supports drag-to-rank for portfolio Owners/Managers, including multi-row moves and rank rebalance maintenance.
- Adds current scoring-model breakdown columns to the ranking board so rank and priority score can be reviewed together.
- Updates docs for the new ranking workflow, API endpoints, background job, and AI/domain references.

## Details

- Projects now receive a fractional rank on creation and expose a derived display position in project list responses.
- Portfolio ranking APIs support moving ranked project batches, manually rebalancing ranks, and loading ranking scoreboard data.
- A `PortfolioRankRebalance` Hangfire job re-spaces dense fractional ranks back to clean whole-number gaps.
- Project list/card views now expose rank context where applicable.
- Docs now describe portfolio ranking, the Ranking tab, score breakdown behavior, permissions, and maintenance behavior.

## Validation

- Added backend application/domain tests for moving and rebalancing portfolio project ranks.
- Added frontend tests for the ranking helper and project card rank sorting.
- Ran Docusaurus docs build successfully with `npm run build` from `docs-site`.

### CODE REVIEW

File: Wayd.Web/src/wayd.web.reactclient/src/components/common/wayd-grid.tsx [hunk 1]
```diff
@@ -109,7 +109,10 @@ const WaydGrid = forwardRef<AgGridReact, ModaGridProps>(
       gridRef.current?.api.exportDataAsCsv()
     }
 
-    const onClearFilters = () => {
+    const onClearSortAndFilters = () => {
+      gridRef.current?.api.applyColumnState({
+        defaultState: { sort: null },
+      })
```
[the remark this answers is not in the document]
Comment (destacey) [reply Wayd.Web/src/wayd.web.reactclient/src/components/common/wayd-grid.tsx:115]: Addressed by clearing both sort and sortIndex when resetting column state, so multi-column sort metadata is fully cleared.

File: Wayd.Services/Wayd.ProjectPortfolioManagement/src/Wayd.ProjectPortfolioManagement.Application/Projects/Dtos/ProjectDetailsDto.cs [hunk 2]
```diff
@@ -130,6 +130,13 @@ public sealed record ProjectDetailsDto
     /// </summary>
     public ScoreSummaryDto? CurrentScore { get; set; }
 
+    /// <summary>
+    /// The project's fractional rank sort key within its portfolio — the same value as the domain
+    /// <c>Project.Rank</c> (opaque; not the displayed position). Null = unranked (sorts last). Used to
+    /// order the ranking board and drive client-side reorder logic; not for display.
+    /// </summary>
```
[the remark this answers is not in the document]
Comment (destacey) [reply Wayd.Services/Wayd.ProjectPortfolioManagement/src/Wayd.ProjectPortfolioManagement.Application/Projects/Dtos/ProjectDetailsDto.cs:137]: Addressed by updating the XML docs to state that every project is ranked from creation, removing the stale nullable/unranked wording.

File: Wayd.Services/Wayd.ProjectPortfolioManagement/src/Wayd.ProjectPortfolioManagement.Application/Projects/Queries/GetProjectsQuery.cs [hunk 3]
```diff
@@ -94,6 +94,25 @@ internal sealed class GetProjectsQueryHandler(IProjectPortfolioManagementDbConte
         var config = ProjectListDto.CreateTypeAdapterConfig(now, _currentUser.GetEmployeeId());
         var projects = await query.ProjectToType<ProjectListDto>(config).ToListAsync(cancellationToken);
 
-        return [.. projects.OrderBy(p => p.Name)];
+        var ordered = projects
+            .OrderBy(p => p.Portfolio.Name)
+            .ThenBy(p => p.Rank)
+            .ThenBy(p => p.Name)
+            .ToList();
```
[the remark this answers is not in the document]
Comment (destacey) [reply Wayd.Services/Wayd.ProjectPortfolioManagement/src/Wayd.ProjectPortfolioManagement.Application/Projects/Queries/GetProjectsQuery.cs:101]: Addressed by calculating positions after ordering by the unique portfolio id instead of portfolio name. I also updated the projection test to cover duplicate portfolio names.

File: Wayd.Infrastructure/src/Wayd.Infrastructure.Migrators.MSSQL/Migrations/20260607161829_Make-Project-Rank-Required.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        migrationBuilder.Sql($@"
+            IF NOT EXISTS (
+                SELECT 1
+                FROM [Auditing].[AuditTrails]
+                WHERE [CorrelationId] <> '{CorrelationId}'
+                  AND [SchemaName] = 'Ppm'
+                  AND [TableName] = 'Projects'
+                  AND [Type] = 'Update'
+                  AND [AffectedColumns] LIKE '%""Rank""%'
+            )
```
[the remark this answers is not in the document]
Comment (destacey) [reply Wayd.Infrastructure/src/Wayd.Infrastructure.Migrators.MSSQL/Migrations/20260607161829_Make-Project-Rank-Required.cs:120]: Addressed by aligning the rollback guard with the singular Project table name written by the Up migration.

File: Wayd.Infrastructure/src/Wayd.Infrastructure.Migrators.MSSQL/Migrations/20260607161829_Make-Project-Rank-Required.cs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                  AND a.[TableName] = 'Projects'
+                  AND a.[Type] = 'Update'
+                  AND a.[AffectedColumns] = '[""Rank""]';
+
+                DELETE FROM [Auditing].[AuditTrails]
+                WHERE [CorrelationId] = '{CorrelationId}'
+                  AND [SchemaName] = 'Ppm'
+                  AND [TableName] = 'Projects'
+                  AND [Type] = 'Update'
+                  AND [AffectedColumns] = '[""Rank""]';
```
[the remark this answers is not in the document]
Comment (destacey) [reply Wayd.Infrastructure/src/Wayd.Infrastructure.Migrators.MSSQL/Migrations/20260607161829_Make-Project-Rank-Required.cs:138]: Addressed by using the same singular Project table-name filter in the rollback update/delete cleanup, so it matches the audit rows inserted by the Up migration.
